pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -118,8 +119,13 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 				copied.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
 				copied.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledNewerDeploymentExists
 
-				updatedDeployment, err := c.rn.ReplicationControllers(copied.Namespace).Update(copied)
-				if err != nil {
+				// Retry faster on conflicts
+				var updatedDeployment *kapi.ReplicationController
+				if err := kclient.RetryOnConflict(wait.Backoff{Steps: 5}, func() error {
+					var err error
+					updatedDeployment, err = c.rn.ReplicationControllers(copied.Namespace).Update(copied)
```

Comment:
not RC listers...

On Thu, Aug 18, 2016 at 5:15 PM, David Eads <notifications@github.com>
wrote:

> In pkg/deploy/controller/deploymentconfig/controller.go
> <https://github.com/openshift/origin/pull/10507#discussion_r75327857>:
>
> > @@ -118,8 +119,13 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
> >  				copied.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
> >  				copied.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledNewerDeploymentExists
> >
> > -				updatedDeployment, err := c.rn.ReplicationControllers(copied.Namespace).Update(copied)
> > -				if err != nil {
> > +				// Retry faster on conflicts
> > +				var updatedDeployment *kapi.ReplicationController
> > +				if err := kclient.RetryOnConflict(wait.Backoff{Steps: 5}, func() error {
> > +					var err error
> > +					updatedDeployment, err = c.rn.ReplicationControllers(copied.Namespace).Update(copied)
>
> @deads2k <https://github.com/deads2k> will need Get() func in cache to
> re-get the rc
>
> Listers already have a get function.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10507/files/095b81adcf1aeb4f3f928f95f9e5fc1ebf91588f#r75327857>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACsaONbQSMzSUok8fR-5utg6oF7wixiks5qhHb-gaJpZM4JnlAW>
> .
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering